File: Tests/Full/C#/AudioTests.cs
using System;
using System.Collections.Generic;
using System.Text;
using SwinGame;

namespace Tests
{

    class AudioTests : IGameTestLoader
    {
        #region IGameTestLoader Members

        public void AddTo(List<TestSuite> list)
        {
            TestSuite result = new TestSuite("Audio Tests");
            result.Add(new PlaySoundEffectTest());
            result.Add(new PlayMusicTest());
            list.Add(result);
        }

        #endregion

        private class PlaySoundEffectTest : TestSet
        {
            private readonly static string METHS =
                "PlaySoundEffect, IsSoundEffectPlaying, StopSoundEffect";

            private readonly static string INST = 
                "pl[a]y sound once" + Environment.NewLine + 
                "play [i]f not playing" + Environment.NewLine + 
                "[s]top sound effect" + Environment.NewLine + 
                "[l]oop sound" + Environment.NewLine +
                "play at 25% [v]olume";

            public PlaySoundEffectTest() : base(METHS, INST){}

            protected override void ToRun(System.Drawing.Rectangle toDrawIn)
            {
                SoundEffect se = GameResources.GameSound("Shock");
                if(Input.WasKeyTyped(Keys.VK_A)) Audio.PlaySoundEffect(se);
		        if(Input.WasKeyTyped(Keys.VK_I))
			        if(false == Audio.IsSoundEffectPlaying(se)) 
                        Audio.PlaySoundEffect(se);
		        if(Input.WasKeyTyped(Keys.VK_S)) Audio.StopSoundEffect(se);
		        if(Input.WasKeyTyped(Keys.VK_L)) Audio.PlaySoundEffect(se, -1);
                if (Input.WasKeyTyped(Keys.VK_V)) se.Play(0.25f);
            }
        }

        private class PlayMusicTest : TestSet
        {
            private readonly static string METHS =
                "PlayMusic, IsMusicPlaying, StopMusic";

            private readonly static string INST =
                "pl[a]y musix once" + Environment.NewLine +
                "play [i]f not playing" + Environment.NewLine +
                "[s]top music" + Environment.NewLine +
                "[l]oop music" + Environment.NewLine +
                "[up/down] to change volume";

            public PlayMusicTest() : base(METHS, INST) { }

            protected override void ToRun(System.Drawing.Rectangle toDrawIn)
            {
                Music se = GameResources.GameMusic("Fast");
                if (Input.WasKeyTyped(Keys.VK_A)) Audio.PlayMusic(se, 1);
                if (Input.WasKeyTyped(Keys.VK_I))
                    if (false == Audio.IsMusicPlaying())
                        Audio.PlayMusic(se);
                if (Input.WasKeyTyped(Keys.VK_S)) Audio.StopMusic();
                if (Input.WasKeyTyped(Keys.VK_L)) Audio.PlayMusic(se);
                if (Input.IsKeyPressed(Keys.VK_UP)) Music.Volume += 0.01f;
                if (Input.IsKeyPressed(Keys.VK_DOWN)) Music.Volume -= 0.01f;


            }
        }
    }
}

File: Tests/Full/Pascal/src/AudioTests.pas
unit AudioTests;

interface
	uses TestFramework;

	procedure AddAudioSuite(var suites: TestSuites); 

implementation
	uses GameResources, sgAudio, sgInput, sgGraphics, sgTypes, sgGeometry, sgText, SysUtils;
	
	procedure TestPlaySound(const drawIn: Rectangle);
	begin
		if KeyTyped(VK_A) then PlaySoundEffect(SoundEffectNamed('Shock'));
		if KeyTyped(VK_I) then 
		begin	
			if false = SoundEffectPlaying(SoundEffectNamed('Shock')) then PlaySoundEffect(SoundEffectNamed('Shock'));
		end;
		if KeyTyped(VK_S) then StopSoundEffect(SoundEffectNamed('Shock'));
		if KeyTyped(VK_L) then PlaySoundEffect(SoundEffectNamed('Shock'), -1);
	end;
	
	procedure TestPlayMusic(const drawIn: Rectangle);
	begin
		if KeyTyped(VK_A) then PlayMusic(MusicNamed('Fast'), 1);
		if KeyTyped(VK_I) then 
		begin	
			if false = MusicPlaying() then PlayMusic(MusicNamed('Fast'));
		end;
		if KeyTyped(VK_S) then StopMusic();
		if KeyTyped(VK_L) then PlayMusic(MusicNamed('Fast'));
		
		if KeyDown(VK_UP) then 
		begin
		  SetMusicVolume(MusicVolume() + 0.01);
		end;
		if KeyDown(VK_DOWN) then 
		begin
		  SetMusicVolume(MusicVolume() - 0.01);
		end;
		
		DrawText('Volume ' + FloatToStr(MusicVolume()), ColorWhite, 100, 100);
	end;
	
	function GetAudioTests(): TestSuite;
	var
		i: Integer;
	begin
		result.Title := 'Audio Tests';
		SetLength(result.Tests, 2);
		
		for i := 0 to High(result.Tests) do
		begin
			InitTest(result.Tests[i]);
		end;
		
		with result.Tests[0] do
		begin
			MethodBeingTested := 'PlaySoundEffect, IsSoundEffectPlaying, StopSoundEffect';
			Instructions := 'pl[a]y sound once' + LineEnding + '' + LineEnding + 'play [i]f not playing' + LineEnding + '[s]top sound effect' + LineEnding + '[l]oop sound';
			ToRun := @TestPlaySound;
		end;
		with result.Tests[1] do
		begin
			MethodBeingTested := 'PlayMusic, MusicPlaying, StopMusic';
			Instructions := 'pl[a]y music once' + LineEnding + 'play [i]f not playing' + LineEnding + '[s]top music' + LineEnding + '[l]oop music';
			ToRun := @TestPlayMusic;
		end;
	end;
	
	procedure AddAudioSuite(var suites: TestSuites);
	begin
		suites[High(suites)] := GetAudioTests();
	end;

end.